extensions/ql-vscode/src/qlpack-generator.ts

```diff
@@ -0,0 +1,85 @@
+import { mkdir, writeFile } from "fs-extra";
+import { dump } from "js-yaml";
+import { join } from "path";
+import { CodeQLCliServer } from "./cli";
+
+export type QueryLanguage =
+  | "csharp"
+  | "cpp"
+  | "go"
+  | "java"
+  | "javascript"
+  | "python"
+  | "ruby"
+  | "swift";
+
+export class QlPackGenerator {
+  private readonly qlpackName: string;
+  private readonly qlpackVersion: string;
+  private readonly header: string;
+  private readonly qlpackFileName: string;
+
+  constructor(
+    private readonly folderName: string,
+    private readonly queryLanguage: QueryLanguage,
+    private readonly cliServer: CodeQLCliServer,
+  ) {
+    this.qlpackName = `getting-started/codeql-extra-queries-${this.queryLanguage}`;
+    this.qlpackVersion = "1.0.0";
+    this.header = "# This is an automatically generated file.\n\n";
+
+    this.qlpackFileName = "qlpack.yml";
+  }
+
+  public async generate() {
+    await mkdir(this.folderName);
+
+    // create qlpack.yml
+    await this.createQlPackYaml();
+
+    // create example.ql
+    await this.createExampleQlFile();
+
+    // create codeql-pack.lock.yml
+    await this.createCodeqlPackLockYaml();
+  }
+
+  private async createQlPackYaml() {
+    const qlPackFile = join(this.folderName, this.qlpackFileName);
+
+    const qlPackYml = {
+      name: this.qlpackName,
+      version: this.qlpackVersion,
+      dependencies: {
+        [`codeql/${this.queryLanguage}-all`]: "*",
+      },
+    };
+
+    await writeFile(qlPackFile, this.header + dump(qlPackYml), "utf8");
+  }
+
+  private async createExampleQlFile() {
+    const exampleQlFile = join(this.folderName, "example.ql");
+
+    const exampleQl = `
+/**
+ * @name Empty block
```

Comment:
Maybe nice to add a comment here that this is auto-generated and perhaps even a url where they can go to find more information?

Reply:
I did add one but was afraid the comment would be significant! I'll add it back.